### PR TITLE
Add component for removing portfolio items

### PR DESCRIPTION
**Pull request by hstastna**
**What:**
Add the component and another smaller changes (syntax, styling, removing button for editing portfolios from navigation, etc).

**Screenshots:**
_Remove products_ button for removing portfolio items from the portfolio: 
remove_button
Selecting and then removing product from the portfolio:
removing
After removing selected product from the portfolio:
removed

### CODE REVIEW

File: src/SmartComponents/Portfolio/PortfolioItems.js [hunk 1]
```diff
@@ -1,32 +1,33 @@
...
+      showItems: '',
+      filteredItems: [],
+      selectItems: [],
+      isLoaded: false,
+    };
 
     fetchData(apiProps) {
         if (apiProps && apiProps.portfolio) {
-            this.props.fetchPortfolioItemsWithPortfolio(apiProps.portfolio);
+            this.props.fetchPortfolioItemsWithPortfolio(apiProps.portfolio).then(() =>
```
Comment (lgalis) [inline src/SmartComponents/Portfolio/PortfolioItems.js]: You can just call this.props.fetchPortfolioItemsWithPortfolio (apiProps.portfolio);  here 
The middleware will dispatch the ActionTypes.FETCH_PORTFOLIO_ITEMS_WITH_PORTFOLIO + '_FULFILLED' action and set the payload to the api result and isLoading to false.

File: src/SmartComponents/Portfolio/PortfolioItems.js [hunk 2]
```diff
@@ -76,12 +126,10 @@ function mapStateToProps(state) {
     };
 }
 
-const mapDispatchToProps = dispatch => {
-    return {
-        fetchPortfolioItems: apiProps => dispatch(fetchPortfolioItems(apiProps)),
-        fetchPortfolioItemsWithPortfolio: apiProps => dispatch(fetchPortfolioItemsWithPortfolio(apiProps))
-    };
-};
+const mapDispatchToProps = dispatch => bindActionCreators({
```
Comment (lgalis) [inline src/SmartComponents/Portfolio/PortfolioItems.js]: What is the reason for this change? Why is the binding necessary?

**Comment by hstastna** [comment]
@miq-bot add_reviewer @Hyperkid123

**Comment by Hyperkid123** [comment]
@hstastna travis is faling on some linting issues. Can you fix them please? These are just cosmetic changes to the code. 

you can run this command from console to list them:
```
yarn eslint ./src --quiet
```
(quiet hides warnings)

File: src/Helpers/Portfolio/PortfolioHelper.js [hunk 3]
```diff
@@ -59,3 +59,16 @@ export async function updatePortfolio(portfolioData) {
...
+
+export async function removePortfolioItem(portfolioItemId) {
+  return userApi.destroyPortfolioItem(portfolioItemId).then(data => {
+    console.log('Remove Portfolio Item Called successfully.');
+    return data;
+  });
+}
+
+export async function removePortfolioItems(portfolioItemIds) {
+  for (const itemId of portfolioItemIds) {
```
Comment (Hyperkid123) [inline src/Helpers/Portfolio/PortfolioHelper.js]: This should be probably Promise.all. We want to wait until all the requests are successfully completed.
 You have to prepare array of promises and resolve them after they are all completed. Code should look something like this:

```javascript
export function removePortfolioItems(portfolioItemIds) {
  return Promise.all(portfolioItemIds.map(async itemId =>  await removePortfolioItem(itemId)))
}

```

File: src/Helpers/Portfolio/PortfolioHelper.js [hunk 4]
```diff
@@ -59,3 +59,16 @@ export async function updatePortfolio(portfolioData) {
 export async function removePortfolio(portfolioId) {
   await userApi.destroyPortfolio(portfolioId);
 }
+
+export async function removePortfolioItem(portfolioItemId) {
+  return userApi.destroyPortfolioItem(portfolioItemId).then(data => {
+    console.log('Remove Portfolio Item Called successfully.');
```
Comment (Hyperkid123) [inline src/Helpers/Portfolio/PortfolioHelper.js]: console log :wink: 

And after you remove it you don't need the `.then` code. Just return the `userApi.destroyPortfolioItem(portfolioItemId)` API call.

File: src/PresentationalComponents/Portfolio/PortfolioActionToolbar.js [hunk 5]
```diff
@@ -56,14 +56,14 @@ class PortfolioActionToolbar extends Component {
...
-                <Button variant="link"  aria-label="Add Products to Portfolio">
+                <Button variant="link" aria-label="Add Products to Portfolio">
                   Add Products
                 </Button>
               </Link>
             </ToolbarItem>
             <ToolbarItem className={ css(spacingStyles.mxLg) }>
-              <Button variant="plain" aria-label="Remove Products from Portfolio">
-                Remove Products
+              <Button variant="plain" aria-label="Remove Products from Portfolio" onClick={this.props.onStartRemovingProducts}>
```
Comment (Hyperkid123) [inline src/PresentationalComponents/Portfolio/PortfolioActionToolbar.js]: missing spaces after `{` and before `}`

File: src/PresentationalComponents/Portfolio/PortfolioActionToolbar.js [hunk 6]
```diff
@@ -80,7 +80,12 @@ PortfolioActionToolbar.propTypes = {
   onClickEditPortfolio: propTypes.func,
   addProductsRoute: propTypes.string.isRequired,
   editPortfolioRoute: propTypes.string.isRequired,
-  removePortfolioRoute: propTypes.string.isRequired
+  removePortfolioRoute: propTypes.string.isRequired,
+  onStartRemovingProducts: propTypes.func
```
Comment (Hyperkid123) [inline src/PresentationalComponents/Portfolio/PortfolioActionToolbar.js]: This sgould be probably required without the default props declaration.

File: src/SmartComponents/Portfolio/Portfolio.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import React, { Component, Fragment } from 'react';
+import React, { Component, Fragment, cloneElement } from 'react';
```
Comment (Hyperkid123) [inline src/SmartComponents/Portfolio/Portfolio.js]: clone element is never used

File: src/SmartComponents/Portfolio/Portfolio.js [hunk 8]
```diff
@@ -41,6 +45,36 @@ class Portfolio extends Component {
     }
   }
 
+  startRemovingProducts = () => {
```
Comment (Hyperkid123) [inline src/SmartComponents/Portfolio/Portfolio.js]: you can do these as single line methods:
```javascript
finishRemovingProducts = () => this.setState({ isRemoving: false });
```
Also applies to `finishRemovingProducts`

File: src/SmartComponents/Portfolio/Portfolio.js [hunk 9]
```diff
@@ -54,18 +88,24 @@ class Portfolio extends Component {
...
             addProductsRoute={ addProductsRoute }
             editPortfolioRoute={ editPortfolioRoute }
             removePortfolioRoute={ removePortfolioRoute }
+            onStartRemovingProducts={ this.startRemovingProducts }
           />
         </div>
       }
       <Route exact path="/portfolio/:id/edit-portfolio" component={ AddPortfolioModal } />
       <Route exact path="/portfolio/:id/remove-portfolio" component={ RemovePortfolioModal } />
+      { this.state.isRemoving &&
```
Comment (Hyperkid123) [inline src/SmartComponents/Portfolio/Portfolio.js]: I would probably hook this component to route. So it works the same way how the adding adding products work:
https://github.com/ManageIQ/service_portal-ui/blob/4fb34d37110c0675a55af0883a98719f5411db88/src/SmartComponents/Portfolio/Portfolio.js#L143

It will make the code a bit cleaner

File: src/SmartComponents/Portfolio/Portfolio.js [hunk 10]
```diff
@@ -41,6 +45,36 @@ class Portfolio extends Component {
...
+    this.setState({ isRemoving: false });
+  }
+
+  removeProducts = () => {
+    const itemIdsToRemove = this.state.selectItems;
+
+    this.finishRemovingProducts();
+
+    removePortfolioItems(itemIdsToRemove).then(() => {
+      this.setState({
```
[the remark this answers is not in the document]
Comment (hstastna) [reply src/SmartComponents/Portfolio/Portfolio.js]: And what about emptying `selectItems` array?

File: src/Helpers/Portfolio/PortfolioHelper.js (hunk 3 above)
Comment (Hyperkid123) [reply src/Helpers/Portfolio/PortfolioHelper.js]: It waits for individual requests, this will wait for everything.

File: src/SmartComponents/Portfolio/Portfolio.js [hunk 11]
```diff
@@ -15,14 +15,17 @@ import AddPortfolioModal from './add-portfolio-modal';
...
-      isKebabOpen: false,
-      isOpen: false,
-      filteredItems: []
-    };
+  state = {
+    portfolioId: '',
+    isKebabOpen: false,
+    isOpen: false,
+    filteredItems: [],
+    selectItems: []
```
Comment (Hyperkid123) [inline src/SmartComponents/Portfolio/Portfolio.js]: type: should be probably `selectedItems`

File: src/SmartComponents/Portfolio/Portfolio.js [hunk 12]
```diff
@@ -78,24 +104,45 @@ class Portfolio extends Component {
...
-    let filteredItems = {
-      items: this.props.portfolioItems.map(item => <PortfolioItem key={ item.id } { ...item }/>),
+    const title = this.props.portfolio ? this.props.portfolio.name : '';
+
+    const filteredItems = {
+      items: this.props.portfolioItems.map(item => (
+        <PortfolioItem
+          key={ item.id }
+          { ...item }
+          { ...{
```
Comment (Hyperkid123) [inline src/SmartComponents/Portfolio/Portfolio.js]: No need for the double object destructing, just assign them as regular props:
```javascript
 isSelectable={ this.props.location.pathname.includes('/remove-products') }
...
```
